server/routes/index.js

```diff
@@ -1,8 +1,86 @@
 const express = require("express");
 const router = express.Router();
+const databaseSettings = require("./knexfile.js").development;
+const knex = require("knex")(databaseSettings);
+const bcrypt = require("bcrypt");
+const jwt = require("jsonwebtoken");
+const SECRET = "authenticate this user please";
 
 router.get("/welcome", function (req, res, next) {
   res.status(200).send({ welcomeMessage: "Step 1 (completed)" });
 });
 
+router.post("/register", function (req, res, next) {
+  const createdUser = knex("users")
+    .where({ username: req.body.username })
+    .select();
+  const password = req.body.password;
+  const email = req.body.email;
+
+  if (createdUser.length > 0) {
+    res.status(400).send({ response: "Username already taken" });
+  } else if (createdUser.length === 0 && password.length >= 6 && email) {
```

Comment:
It would be nice if we also validated the password length and email and sent the appropriate error back